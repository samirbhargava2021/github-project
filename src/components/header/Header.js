import React from "react";
import "./Header.scss";
import { CMS } from '../../cms/CMS';

const Header = () => {
  return (
    <header className="header">
      <h1 className="header__title">{CMS.searchRepos}</h1>
    </header>
  );
};

export default Header;