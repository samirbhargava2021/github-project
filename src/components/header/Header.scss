.header {
  padding: 0 20px;
  background-color: #140350;
  height: 80px;
  display: flex;
  align-items: center;
  &__title {
    color: #fff;
    margin: 0;
    font-size: 24px;
  }
}
