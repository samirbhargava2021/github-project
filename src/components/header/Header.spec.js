import { shallow } from "enzyme";
import { render } from "@testing-library/react";
import Header from "./Header";

describe("HeaderComponent", () => {
  it("should render Header Component", () => {
    const component = shallow(<Header />);
    expect(component.getElements()).toMatchSnapshot();
  });
  it("should have className header to load the header with proper styling", () => {
    const { container } = render(<Header />);
    expect(container.firstChild.classList.contains('header')).toBe(true)
  });
  it("should have text 'Search Repos'", () => {
    const { getByText } = render(<Header />);
    expect(getByText("Search Repos")).toBeDefined();
  });
});
