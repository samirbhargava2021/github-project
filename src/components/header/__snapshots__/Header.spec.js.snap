// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderComponent should render Header Component 1`] = `
Array [
  <header
    className="header"
  >
    <h1
      className="header__title"
    >
      Search Repos
    </h1>
  </header>,
]
`;
